### PR TITLE
Audit Persistence Abstractions

### PR DESCRIPTION
**Pull request by jpalac**
Added RavenDb abstractions for API methods
Adjusted API tests

Next step is to add the inmemory and sql abstractions, move all abstractions to their own projects and load them dynamically

### CODE REVIEW

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
```
Comment (HEskandari) [inline src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:57]: Do we need these?

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            {
+                var rows = await connection.QueryAsync("SELECT * FROM [MessagesView]").ConfigureAwait(false);
+
+                foreach (var row in rows)
+                {
+                    knownEndpoints.Add(new KnownEndpointsView
+                    {
+                        Id = row.Id,
+                        HostDisplayName = row.HostDisplayName
+                        //EndpointDetails = row.EndpointDetails
```
Comment (HEskandari) [inline src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:79]: Remove?

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                        MessageType = row.MessageType,
+                        ProcessedAt = row.ProcessedAt,
+                        ProcessingTime = row.ProcessingTime,
+                        Status = row.Status,
+                        TimeSent = row.TimeSent
+                        //Headers = row.Headers,
+                        //InvokedSagas = row.InvokedSagas,
+                        //OriginatesFromSaga = row.OriginatesFromSaga,
+                        //ReceivingEndpoint = row.ReceivingEndpoint,
+                        //SendingEndpoint = row.SendingEndpoint
```
Comment (HEskandari) [inline src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:124]: Remove?

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
```
Comment (HEskandari) [inline src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:168]: Same as above

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs (hunk 2 above)
Comment (jpalac) [reply src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:79]: It's what's left to do so I've added TODO comments above those to make it clear

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs (hunk 3 above)
Comment (jpalac) [reply src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:124]: It's what's left to do so I've added TODO comments above those to make it clear

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs (hunk 1 above)
Comment (jpalac) [reply src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:58]: It's what's left to do so I've added TODO comments above those to make it clear

File: src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs (hunk 4 above)
Comment (jpalac) [reply src/ServiceControl.Audit.Persistence.SqlServer/SqlDbAuditDataStore.cs:172]: It's what's left to do so I've added TODO comments above those to make it clear

**Review by HEskandari — APPROVED** [review]

**Pull request merged**